pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -79,11 +103,88 @@ func (proxier *metaProxier) OnServiceUpdate(oldService, service *v1.Service) {
 	if utilproxy.ShouldSkipService(service) {
 		return
 	}
-	// IPFamily is immutable, hence we only need to check on the new service
-	if utilnet.IsIPv6String(service.Spec.ClusterIP) {
-		proxier.ipv6Proxier.OnServiceUpdate(oldService, service)
-	} else {
-		proxier.ipv4Proxier.OnServiceUpdate(oldService, service)
+	// case zero: this allows skew between new proxy and old apiserver
+	if len(service.Spec.IPFamilies) == 0 {
+		actual := proxier.getProxyByClusterIP(oldService)
```

Comment:
can a Service change it's ClusterIP from one family to another? should we use `service` instead of `oldService`?

Reply:
ClusterIPs are immutable. Only upgrade (one ClusterIP=> two ClusterIPs) and downgrade (Two ClusterIPs => one ClusterIP). But user can not change IPs.

Reply:
and is it possible to have a service with ClusterIPs = [ "10.2.2.2","fd00:21:2::2"| that moves to ClusterIPs["fd00:21:2::2"] ?
 the new ClusterIP will be different than the old ClusterIP

Reply:
No. This is not allowed.